Servatrice failing to compile in Windows

I am seeing linkage errors when attempting to compile servatrice using Windows with Qt5.  The errors are being thrown about the updated SMTP components.  I would guess that the issue was introduced with https://github.com/Cockatrice/Cockatrice/commit/1ffc9b45612a4f4a45ccb4facbdf1d64d977b9da .

The server seems to compile fine under linux.

```
4>c:\cockatrice\build\servatrice\moc_qxtsmtp.cpp(74): warning C4273: 'QxtSmtp::qt_static_metacall' : inconsistent dll linkage
4>          c:\cockatrice\build\servatrice\../../servatrice/src/smtp/qxtsmtp.h(41) : see previous definition of 'qt_static_metacall'
4>c:\cockatrice\build\servatrice\moc_qxtsmtp.cpp(102): warning C4273: 'staticMetaObjectExtraData' : inconsistent dll linkage
4>          c:\cockatrice\build\servatrice\../../servatrice/src/smtp/qxtsmtp.h(41) : see previous definition of 'private: static QMetaObjectExtraData const QxtSmtp::staticMetaObjectExtraData'
4>c:\cockatrice\build\servatrice\moc_qxtsmtp.cpp(102): error C2491: 'QxtSmtp::staticMetaObjectExtraData' : definition of dllimport static data member not allowed
4>c:\cockatrice\build\servatrice\moc_qxtsmtp.cpp(106): warning C4273: 'staticMetaObject' : inconsistent dll linkage
4>          c:\cockatrice\build\servatrice\../../servatrice/src/smtp/qxtsmtp.h(41) : see previous definition of 'public: static QMetaObject const QxtSmtp::staticMetaObject'
4>c:\cockatrice\build\servatrice\moc_qxtsmtp.cpp(106): error C2491: 'QxtSmtp::staticMetaObject' : definition of dllimport static data member not allowed
4>c:\cockatrice\build\servatrice\moc_qxtsmtp.cpp(116): warning C4273: 'QxtSmtp::metaObject' : inconsistent dll linkage
4>          c:\cockatrice\build\servatrice\../../servatrice/src/smtp/qxtsmtp.h(41) : see previous definition of 'metaObject'
4>c:\cockatrice\build\servatrice\moc_qxtsmtp.cpp(121): warning C4273: 'QxtSmtp::qt_metacast' : inconsistent dll linkage
4>          c:\cockatrice\build\servatrice\../../servatrice/src/smtp/qxtsmtp.h(41) : see previous definition of 'qt_metacast'
4>c:\cockatrice\build\servatrice\moc_qxtsmtp.cpp(129): warning C4273: 'QxtSmtp::qt_metacall' : inconsistent dll linkage
4>          c:\cockatrice\build\servatrice\../../servatrice/src/smtp/qxtsmtp.h(41) : see previous definition of 'qt_metacall'
4>c:\cockatrice\build\servatrice\moc_qxtsmtp.cpp(143): warning C4273: 'QxtSmtp::connected' : inconsistent dll linkage
4>          c:\cockatrice\build\servatrice\../../servatrice/src/smtp/qxtsmtp.h(86) : see previous definition of 'connected'
4>c:\cockatrice\build\servatrice\moc_qxtsmtp.cpp(149): warning C4273: 'QxtSmtp::connectionFailed' : inconsistent dll linkage
4>          c:\cockatrice\build\servatrice\../../servatrice/src/smtp/qxtsmtp.h(87) : see previous definition of 'connectionFailed'
4>c:\cockatrice\build\servatrice\moc_qxtsmtp.cpp(155): warning C4273: 'QxtSmtp::connectionFailed' : inconsistent dll linkage
4>          c:\cockatrice\build\servatrice\../../servatrice/src/smtp/qxtsmtp.h(88) : see previous definition of 'connectionFailed'
4>c:\cockatrice\build\servatrice\moc_qxtsmtp.cpp(162): warning C4273: 'QxtSmtp::encrypted' : inconsistent dll linkage
4>          c:\cockatrice\build\servatrice\../../servatrice/src/smtp/qxtsmtp.h(89) : see previous definition of 'encrypted'
4>c:\cockatrice\build\servatrice\moc_qxtsmtp.cpp(168): warning C4273: 'QxtSmtp::encryptionFailed' : inconsistent dll linkage
4>          c:\cockatrice\build\servatrice\../../servatrice/src/smtp/qxtsmtp.h(90) : see previous definition of 'encryptionFailed'
4>c:\cockatrice\build\servatrice\moc_qxtsmtp.cpp(174): warning C4273: 'QxtSmtp::encryptionFailed' : inconsistent dll linkage
4>          c:\cockatrice\build\servatrice\../../servatrice/src/smtp/qxtsmtp.h(91) : see previous definition of 'encryptionFailed'
4>c:\cockatrice\build\servatrice\moc_qxtsmtp.cpp(181): warning C4273: 'QxtSmtp::authenticated' : inconsistent dll linkage
4>          c:\cockatrice\build\servatrice\../../servatrice/src/smtp/qxtsmtp.h(92) : see previous definition of 'authenticated'
4>c:\cockatrice\build\servatrice\moc_qxtsmtp.cpp(187): warning C4273: 'QxtSmtp::authenticationFailed' : inconsistent dll linkage
4>          c:\cockatrice\build\servatrice\../../servatrice/src/smtp/qxtsmtp.h(93) : see previous definition of 'authenticationFailed'
4>c:\cockatrice\build\servatrice\moc_qxtsmtp.cpp(193): warning C4273: 'QxtSmtp::authenticationFailed' : inconsistent dll linkage
4>          c:\cockatrice\build\servatrice\../../servatrice/src/smtp/qxtsmtp.h(94) : see previous definition of 'authenticationFailed'
4>c:\cockatrice\build\servatrice\moc_qxtsmtp.cpp(200): warning C4273: 'QxtSmtp::senderRejected' : inconsistent dll linkage
4>          c:\cockatrice\build\servatrice\../../servatrice/src/smtp/qxtsmtp.h(96) : see previous definition of 'senderRejected'
4>c:\cockatrice\build\servatrice\moc_qxtsmtp.cpp(207): warning C4273: 'QxtSmtp::senderRejected' : inconsistent dll linkage
4>          c:\cockatrice\build\servatrice\../../servatrice/src/smtp/qxtsmtp.h(97) : see previous definition of 'senderRejected'
4>c:\cockatrice\build\servatrice\moc_qxtsmtp.cpp(214): warning C4273: 'QxtSmtp::recipientRejected' : inconsistent dll linkage
4>          c:\cockatrice\build\servatrice\../../servatrice/src/smtp/qxtsmtp.h(98) : see previous definition of 'recipientRejected'
4>c:\cockatrice\build\servatrice\moc_qxtsmtp.cpp(221): warning C4273: 'QxtSmtp::recipientRejected' : inconsistent dll linkage
4>          c:\cockatrice\build\servatrice\../../servatrice/src/smtp/qxtsmtp.h(99) : see previous definition of 'recipientRejected'
4>c:\cockatrice\build\servatrice\moc_qxtsmtp.cpp(228): warning C4273: 'QxtSmtp::mailFailed' : inconsistent dll linkage
4>          c:\cockatrice\build\servatrice\../../servatrice/src/smtp/qxtsmtp.h(100) : see previous definition of 'mailFailed'
4>c:\cockatrice\build\servatrice\moc_qxtsmtp.cpp(235): warning C4273: 'QxtSmtp::mailFailed' : inconsistent dll linkage
4>          c:\cockatrice\build\servatrice\../../servatrice/src/smtp/qxtsmtp.h(101) : see previous definition of 'mailFailed'
4>c:\cockatrice\build\servatrice\moc_qxtsmtp.cpp(242): warning C4273: 'QxtSmtp::mailSent' : inconsistent dll linkage
4>          c:\cockatrice\build\servatrice\../../servatrice/src/smtp/qxtsmtp.h(102) : see previous definition of 'mailSent'
4>c:\cockatrice\build\servatrice\moc_qxtsmtp.cpp(249): warning C4273: 'QxtSmtp::finished' : inconsistent dll linkage
4>          c:\cockatrice\build\servatrice\../../servatrice/src/smtp/qxtsmtp.h(104) : see previous definition of 'finished'
4>c:\cockatrice\build\servatrice\moc_qxtsmtp.cpp(255): warning C4273: 'QxtSmtp::disconnected' : inconsistent dll linkage
4>          c:\cockatrice\build\servatrice\../../servatrice/src/smtp/qxtsmtp.h(105) : see previous definition of 'disconnected'
```

--- a/servatrice/src/smtp/qxtglobal.h
+++ b/servatrice/src/smtp/qxtglobal.h
@@ -30,6 +30,7 @@
 
 #define QXT_VERSION 0x000602
 #define QXT_VERSION_STR "0.6.2"
+#define QXT_STATIC
 
 //--------------------------global macros------------------------------
 

--- a/servatrice/src/smtp/qxtmailattachment.h
+++ b/servatrice/src/smtp/qxtmailattachment.h
@@ -34,7 +34,7 @@
 #include <QSharedDataPointer>
 #include <QIODevice>
 
-class QxtMailAttachmentPrivate;
+struct QxtMailAttachmentPrivate;
 class QXT_NETWORK_EXPORT QxtMailAttachment
 {
 public:

